client/gefyra/api/bridge.py

```diff
@@ -162,20 +165,31 @@ def bridge(
     # block until all bridges are in place
     #
     logger.info("Waiting for the bridge(s) to become active")
-    from kubernetes.watch import Watch
 
-    w = Watch()
     bridges = {str(ireq["metadata"]["uid"]): False for ireq in ireqs}
-    for event in w.stream(
-        config.K8S_CORE_API.list_namespaced_event, namespace=config.NAMESPACE
-    ):
-        if event["object"].reason == "Established":
-            for ireq in ireqs:
-                if ireq["metadata"]["uid"] == event["object"].involved_object.uid:
-                    bridges[str(ireq["metadata"]["uid"])] = True
-                    logger.info(f"Bridge {ireq['metadata']['name']} established")
-                    if all(bridges.values()):
-                        return True
+    waiting_time = 0
+    # timeout = 0  means no timeout
+    if timeout:
+        waiting_time = timeout
+    while True:
+        # watch whether all relevant bridges have been established
+        kube_ireqs = get_all_interceptrequests(config)
```

Comment:
this is waiting for all ireqs in the cluster what is likely to become a problem in the future with multi peers. why don't you just wait for the bridges that just have been created (ref from above)?